lib/Backend/arm64/LowerMD.cpp

```diff
@@ -7108,12 +7108,31 @@ LowererMD::GenerateFastInlineBuiltInMathRound(IR::Instr* instr)
     IR::Opnd * pointFive = IR::MemRefOpnd::New(m_func->GetThreadContextInfo()->GetDoublePointFiveAddr(), IRType::TyFloat64, this->m_func, IR::AddrOpndKindDynamicDoubleRef);
     this->m_lowerer->InsertAdd(false, floatOpnd, floatOpnd, pointFive, instr);
 
+    // MSR FPSR, xzr
+    IR::Instr* setFPSRInstr = IR::Instr::New(Js::OpCode::MSR_FPSR, instr->m_func);
+    setFPSRInstr->SetSrc1(IR::RegOpnd::New(nullptr, RegZR, TyUint32, instr->m_func));
+    instr->InsertBefore(setFPSRInstr);
+
     // FCVTM intOpnd, floatOpnd
     IR::Opnd * intOpnd = IR::RegOpnd::New(TyInt32, this->m_func);
     instr->InsertBefore(IR::Instr::New(Js::OpCode::FCVTM, intOpnd, floatOpnd, instr->m_func));
 
-    // CBNZ intOpnd, done
-    IR::BranchInstr * cbnzInstr = IR::BranchInstr::New(Js::OpCode::CBNZ, doneLabel, instr->m_func);
+    // MRS exceptReg, FPSR
+    IR::Opnd * exceptReg = IR::RegOpnd::New(TyUint32, this->m_func);
+    instr->InsertBefore(IR::Instr::New(Js::OpCode::MRS_FPSR, exceptReg, instr->m_func));
+
+    IR::LabelInstr* checkOverflowLabel = IR::LabelInstr::New(Js::OpCode::Label, this->m_func);
+
+    // CBNZ intOpnd, done/checkOverflow
+    IR::BranchInstr * cbnzInstr = nullptr;
+    if (instr->GetDst()->IsInt32())
```

Comment:
>if (instr->GetDst()->IsInt32()) [](start = 4, length = 31)

On top we have assert that dst needs to be Int32?

Reply:
I didn't see that. I'll remove the IsInt32 checks.